helper/helper_test.go

```diff
@@ -1,10 +1,166 @@
 package helper
 
 import (
+	"bufio"
+	"fmt"
+	"os"
+
 	"github.com/greenplum-db/gpbackup/utils"
+	"golang.org/x/sys/unix"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+
+	"github.com/greenplum-db/gpbackup/toc"
+	"github.com/pkg/errors"
+)
+
+var (
+	testDir                 = "/tmp/helper_test/20180101/20180101010101"
+	testTocFile             = fmt.Sprintf("%s/test_toc.yaml", testDir)
+	examplePluginTestConfig = "/tmp/test_example_plugin_config.yaml"
+)
+
+type RestoreReaderTestImpl struct{}
+
+func (r *RestoreReaderTestImpl) waitForPlugin() error {
+	// No plugins yet, no errors to detect
+	return nil
+}
+
+func (r *RestoreReaderTestImpl) positionReader(pos uint64, oid int) error {
+	return nil
+}
+
+func (r *RestoreReaderTestImpl) copyData(num int64) (int64, error) {
+	return 1, nil
+}
+
+func (r *RestoreReaderTestImpl) copyAllData() (int64, error) {
+	return 1, nil
+}
+
+func (r *RestoreReaderTestImpl) getFileHandle() *os.File {
+	return nil
+}
+
+func (r *RestoreReaderTestImpl) getReaderType() ReaderType {
+	return "nil"
+}
+
+type SkipFileTestStep struct {
+	getRestorePipeWriterArgExpect string
+	getRestorePipeWriterResult    bool
+	checkSkipFileArgTableOid      int
+	checkSkipFileResult           bool
+}
+
+type RestoreMockHelperImpl struct {
+	stepNo           int
+	expectedOidBatch []oidWithBatch
+	expectedSteps    []SkipFileTestStep
+
+	openedPipesMap map[string]string // Ginkgo matcher works over map value, will diplicate key here
+	restoreData    *RestoreReaderTestImpl
+}
+
+func (h *RestoreMockHelperImpl) openedPipes() []string {
+	if h.openedPipesMap == nil {
+		h.openedPipesMap = make(map[string]string)
+
+		for k := range pipesMap {
+			h.openedPipesMap[k] = k
+		}
+	}
+	ret := make([]string, 0, len(h.openedPipesMap))
+	for k := range h.openedPipesMap {
+		ret = append(ret, k)
+	}
+	return ret
+}
+
+func (h *RestoreMockHelperImpl) getCurStep() SkipFileTestStep {
+	Expect(h.stepNo).To(BeNumerically("<", len(h.expectedSteps)))
+	return h.expectedSteps[h.stepNo]
+}
+
+func NewSkipFileTest(batches []oidWithBatch, steps []SkipFileTestStep) *RestoreMockHelperImpl {
+	var ret = new(RestoreMockHelperImpl)
+	ret.expectedOidBatch = batches
+	ret.expectedSteps = steps
+	ret.openedPipesMap = nil
+	ret.restoreData = &RestoreReaderTestImpl{}
+
+	return ret
+}
```

Comment:
All this only applies to skip file tests, but what about testing everything else?

Reply:
This was started as a method to skip test files, and suggested a method how to test code with omitted most of dependencies. It's was not intended to test "everything else", instead we will suggest a method how "everything else" developer can add unit tests for his code with a smaller efforts.

Reply:
Can you provide examples for other tests (not skip files) with the new mock interfaces?

Reply:
Basically "successfully restores using multiple data files when inputs are valid and no errors occur" has no relation to skip files, so it can be such an example. 
To evaluate the effort it can be valuable to try to make such a test by yourself and show up questions and difficulties.